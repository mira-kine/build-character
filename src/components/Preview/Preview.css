.phrase-list {
  color: black;
  font-weight: 500;
}
