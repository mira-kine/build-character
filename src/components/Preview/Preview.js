import React from 'react';
import Character from '../Character/Character';
import { useState } from 'react';
import Catchphrase from '../Catchphrase/Catchphrase';

import './Preview.css';

export default function Preview() {
  const [head, setHead] = useState('bird-head');
  const [middle, setMiddle] = useState('blue-middle');
  const [bottom, setBottom] = useState('blue-pants');
  const [phrase, setPhrase] = useState('');
  const [phraseList, setPhraseList] = useState([]);
  const [headCount, setHeadCount] = useState(0);
  const [middleCount, setMiddleCount] = useState(0);
  const [bottomCount, setBottomCount] = useState(0);

  return (
    <>
      <div className="character">
        <Character
          {...{
            head,
            setHead,
            middle,
            setMiddle,
            bottom,
            setBottom,
            headCount,
            setHeadCount,
            middleCount,
            setMiddleCount,
            bottomCount,
            setBottomCount,
          }}
        />
        <Catchphrase
          {...{
            phrase,
            setPhrase,
            phraseList,
            setPhraseList,
          }}
        />
        {!!phraseList.length && (
          <div className="phrase-list">
            Your character has said:
            {phraseList.map((elem) => (
              <p key={elem}>{elem}</p>
            ))}
            <div className="parts">
              <img alt={head} src={`${process.env.PUBLIC_URL}/assets/${head}.png`} />
              <img alt={middle} src={`${process.env.PUBLIC_URL}/assets/${middle}.png`} />
              <img alt={bottom} src={`${process.env.PUBLIC_URL}/assets/${bottom}.png`} />
            </div>
            <div className="counter">
              {!!headCount && <p>You have changed the head {headCount} times. </p>}
              {!!middleCount && <p>You have changed the middle {middleCount} times. </p>}
              {!!bottomCount && <p>You have changed the bottom {bottomCount} times. </p>}
            </div>
          </div>
        )}
      </div>
    </>
  );
}
