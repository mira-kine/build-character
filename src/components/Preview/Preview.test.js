import { render } from '@testing-library/react';
import Preview from './Preview';

test('renders the Preview display', () => {
  const container = render(<Preview head="no-head" middle="blue-middle" bottom="blue-pants" />);
  expect(container).toMatchSnapshot();
});
