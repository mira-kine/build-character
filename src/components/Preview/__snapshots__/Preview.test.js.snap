// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Preview display 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="character-editor"
        >
          <div
            class="form-control"
          >
            <select>
              <option
                value="bird-head"
              >
                bird
              </option>
              <option
                value="duck-head"
              >
                duck
              </option>
              <option
                value="dog-head"
              >
                dog
              </option>
              <option
                value="horse-head"
              >
                horse
              </option>
            </select>
          </div>
          <div
            class="form-control"
          >
            <select>
              <option
                value="blue-middle"
              >
                blue
              </option>
              <option
                value="dress-middle"
              >
                dress
              </option>
              <option
                value="pink-middle"
              >
                pink
              </option>
              <option
                value="red-middle"
              >
                red
              </option>
            </select>
          </div>
          <div
            class="form-control"
          >
            <select>
              <option
                value="blue-pants"
              >
                blue pants
              </option>
              <option
                value="dog-pants"
              >
                dog pants
              </option>
              <option
                value="leg-pants"
              >
                leg pants
              </option>
              <option
                value="white-pants"
              >
                white pants
              </option>
            </select>
          </div>
        </div>
        <div
          class="parts"
        >
          <img
            alt="bird-head"
            src="/assets/bird-head.png"
          />
          <img
            alt="blue-middle"
            src="/assets/blue-middle.png"
          />
          <img
            alt="blue-pants"
            src="/assets/blue-pants.png"
          />
        </div>
        <div
          class="catchphrase"
        >
          <div
            class="form-control"
          >
            <input
              style="height: 250px;"
              type="text"
              value=""
            />
          </div>
          <button>
            Save
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="character-editor"
      >
        <div
          class="form-control"
        >
          <select>
            <option
              value="bird-head"
            >
              bird
            </option>
            <option
              value="duck-head"
            >
              duck
            </option>
            <option
              value="dog-head"
            >
              dog
            </option>
            <option
              value="horse-head"
            >
              horse
            </option>
          </select>
        </div>
        <div
          class="form-control"
        >
          <select>
            <option
              value="blue-middle"
            >
              blue
            </option>
            <option
              value="dress-middle"
            >
              dress
            </option>
            <option
              value="pink-middle"
            >
              pink
            </option>
            <option
              value="red-middle"
            >
              red
            </option>
          </select>
        </div>
        <div
          class="form-control"
        >
          <select>
            <option
              value="blue-pants"
            >
              blue pants
            </option>
            <option
              value="dog-pants"
            >
              dog pants
            </option>
            <option
              value="leg-pants"
            >
              leg pants
            </option>
            <option
              value="white-pants"
            >
              white pants
            </option>
          </select>
        </div>
      </div>
      <div
        class="parts"
      >
        <img
          alt="bird-head"
          src="/assets/bird-head.png"
        />
        <img
          alt="blue-middle"
          src="/assets/blue-middle.png"
        />
        <img
          alt="blue-pants"
          src="/assets/blue-pants.png"
        />
      </div>
      <div
        class="catchphrase"
      >
        <div
          class="form-control"
        >
          <input
            style="height: 250px;"
            type="text"
            value=""
          />
        </div>
        <button>
          Save
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
